x-pack/test/apm_api_integration/trial/tests/services/transaction_groups_charts.ts

```diff
@@ -0,0 +1,150 @@
+      after(() => esArchiver.unload(archiveName));
+
+      describe('and fetching transaction groups charts with uiFilters', () => {
+        const serviceName = 'opbeans-java';
+        let response: PromiseReturnType<typeof supertest.get>;
+
+        describe('without uiFilters', () => {
+          before(async () => {
+            response = await supertest.get(
+              `/api/apm/services/${serviceName}/transaction_groups/charts?start=${start}&end=${end}&transactionType=${transactionType}&`
+            );
+          });
+          it('should return an error response', () => {
+            expect(response.status).to.eql(400);
+          });
+        });
+
+        describe('with empty uiFilters', () => {
+          const uiFilters = encodeURIComponent(JSON.stringify({}));
+          before(async () => {
+            response = await supertest.get(
+              `/api/apm/services/${serviceName}/transaction_groups/charts?start=${start}&end=${end}&transactionType=${transactionType}&uiFilters=${uiFilters}`
+            );
+          });
+
+          it('should have a successful response', () => {
+            expect(response.status).to.eql(200);
+          });
+
+          it('should include an ML job id', () => {
+            expect(response.body).to.have.property('anomalyTimeseries');
+            expect(response.body.anomalyTimeseries).to.have.property('jobId');
+            expectSnapshot(response.body.anomalyTimeseries.jobId).toMatchInline(
+              `"apm-environment_not_defined-7ed6-high_mean_transaction_duration"`
+            );
+          });
+
+          it('should return the correct anomaly boundaries', () => {
+            expect(response.body).to.have.property('anomalyTimeseries');
+            expect(response.body.anomalyTimeseries).to.have.property('anomalyBoundaries');
+            expect(response.body.anomalyTimeseries.anomalyBoundaries.length).to.equal(0);
+          });
+        });
+
+        describe('with uiFilters environment defined', () => {
+          const uiFilters = encodeURIComponent(JSON.stringify({ environment: 'production' }));
+          before(async () => {
+            response = await supertest.get(
+              `/api/apm/services/${serviceName}/transaction_groups/charts?start=${start}&end=${end}&transactionType=${transactionType}&uiFilters=${uiFilters}`
+            );
+          });
+
+          it('should have a successful response', () => {
+            expect(response.status).to.eql(200);
+          });
+
+          it('should include an ML job id', () => {
+            expect(response.body).to.have.property('anomalyTimeseries');
+            expect(response.body.anomalyTimeseries).to.have.property('jobId');
+            expectSnapshot(response.body.anomalyTimeseries.jobId).toMatchInline(
+              `"apm-production-229a-high_mean_transaction_duration"`
+            );
+          });
+
+          it('should return the correct anomaly boundaries', () => {
```

Comment:
You use the phrase `should return the correct anomaly boundaries` for multiple different tests to assert different things. Can we omit the term "correct" and instead specify what is returns? Eg `it('returns a non-empty anomaly series')`